### PR TITLE
Issue #2521: Restore support for 'path' to theme_image().

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/2521

Replaces https://github.com/backdrop/backdrop/pull/1786.

### CODE REVIEW

File: core/includes/theme.inc [hunk 1]
```diff
@@ -1927,12 +1927,35 @@ function theme_dropbutton_wrapper($variables) {
...
 
 /**
+ * Preprocess variables for theme_image().
+ *
+ * @since 1.8.0 Function added. Populates $variables['attributes'] before
+ *   theme_image() is called.
+ */
+function template_preprocess_image(&$variables) {
+  // Accept either URI or path key. But both are handled by file_create_url().
+  $path_or_uri = $variables['uri'] ? $variables['uri'] : $variables['path'];
```
Comment (klonos) [inline core/includes/theme.inc:1937]: ...newbie question: should we be checking the case when none of these two variables exist here?

File: core/includes/theme.inc [hunk 2]
```diff
@@ -1948,18 +1971,14 @@ function theme_dropbutton_wrapper($variables) {
...
-  $attributes['src'] = file_create_url($variables['uri']);
-
-  foreach (array('width', 'height', 'alt', 'title') as $key) {
-    if (isset($variables[$key])) {
-      $attributes[$key] = $variables[$key];
-    }
-  }
-
-  return '<img' . backdrop_attributes($attributes) . ' />';
+  return '<img' . backdrop_attributes($variables['attributes']) . ' />';
```
Comment (klonos) [inline core/includes/theme.inc:1981]: Shouldn't there be a space after `'<img'`?

Comment (quicksketch) [reply core/includes/theme.inc:1981]: backdrop_attributes() intentionally includes a prefix space (that way if no attributes were added you don't have an empty space in the tag).

https://api.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_attributes/1

File: core/includes/theme.inc (hunk 1 above)
Comment (quicksketch) [reply core/includes/theme.inc:1937]: Both `uri` and `path` will exist in the `$variables` array thanks to `backdrop_common_theme()` where `theme_image` is registered. Both will have a value of `NULL` though. Ultimately this just means that if neither are passed in, you'll end up with ``, which probably won't be what you want but would be the same result as missing any other theme parameters throughout Backdrop.

**Pull request merged**